### PR TITLE
Add CONTRIBUTING.md

### LINKED ISSUE #1753: Add CONTRIBUTING.md file
It would be helpful for new (and existing) contributors to have a starting point for contributing.

Here's what I'd like to see there:

- [x] Rules for merging PRs (how many people to approve a typical PR)
- [x] JDK 11 requirement
- [x] How to build everything locally to avoid wasting CI time
- [x] [fs2-dev gitter](https://gitter.im/functional-streams-for-scala/fs2-dev)

Feel free to mention anything you'd also like to see! I'll start working on these once we have a consensus on what should go into the file.

### PR DESCRIPTION
**Pull request by kubukoz**
Closes #1753

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+If you run into any problems with tests, binary compatibility or other issues, feel free to ask questions on [the Gitter channel][fs2-dev].
+
+### Website
+
+To build the microsite, use the `microsite/makeMicrosite` task in sbt. Then, go to `site/target/site`, and run a HTTP server on the port of your choice:
+
+```bash
+sbt:root> microsite/makeMicrosite
+info: Compiled in 24s (0 errors, 0 warnings)
+[info] Configuration file: /Users/contributor/dev/fs2/site/target/scala-2.13/resource_managed/main/jekyll/_config.yml
```
Comment (pauljamescleary) [inline CONTRIBUTING.md]: are any other developer requirements necessary (ruby / jekyll / etc)?

Curious as I have everything on my laptop but not sure what else is needed for microsite

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+sbt:root>
+```
+
+Inside the shell, you can compile the sources for the currently selected Scala version using the `compile` command.
+To compile the code for all Scala versions enabled in the build, use `+compile`. To include tests, `test:compile` or `+test:compile`, accordingly.
+
+
+### Testing
+
+To test the code, you can run the `testJVM` or `testJS` command, depending on the platform you want to run the tests on.
```
Comment (pauljamescleary) [inline CONTRIBUTING.md]: are there commands to run _all_ tests, or do you _have_ to run `testJVM` _and_ `testJS`?

File: CONTRIBUTING.md (hunk 1 above)
Comment (kubukoz) [reply CONTRIBUTING.md]: Yeah, pretty sure we'll also need jekyll to build the microsite. I'll mention that.

Comment (kubukoz) [reply CONTRIBUTING.md]: I'll try building it in docker on a clean image and see what else is missing :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (kubukoz) [reply CONTRIBUTING.md]: You can just do `test`. I'll mention that...

File: CONTRIBUTING.md (hunk 1 above)
Comment (kubukoz) [reply CONTRIBUTING.md]: okay, basically we were missing jekyll and node (for running scalajs).

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+[info]       Generating...
+[info]                     done in 0.592 seconds.
+[info]  Auto-regeneration: disabled. Use --watch to enable.
+[success] Total time: 29 s, completed Jan 24, 2020, 2:36:31 PM
+sbt:root> exit
+[info] shutting down sbt server
+
+$ cd site/target/site
+$ python -m SimpleHTTPServer 4000 . # Any other server would do
+Serving HTTP on 0.0.0.0 port 8080 ...
```
Comment (kubukoz) [inline CONTRIBUTING.md]: Port doesn't match command.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+[info] 	   io
+[info] 	   microsite
+[info] 	   reactiveStreams
+[info] 	 * root
+```
+
+Before submitting a change for review, it's worth running some extra checks that will be triggered in Continuous Integration:
+
+```sbt
+sbt:root> fmtCheck; testJVM; testJS; doc; mimaReportBinaryIssues; docs/mdoc; microsite/makeMicrosite
```
Comment (kubukoz) [inline CONTRIBUTING.md]:
```suggestion
sbt:root> fmtCheck; test; doc; mimaReportBinaryIssues; docs/mdoc; microsite/makeMicrosite
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+That will check the formatting, run all tests on the JVM and JS platforms, report any binary compatibility issues (as detected by [MiMa][mima]) and build the site.
```
Comment (kubukoz) [inline CONTRIBUTING.md]:
```suggestion
That will check the formatting, run all tests on the supported platforms, report any binary compatibility issues (as detected by [MiMa][mima]) and build the site.
```

File: CONTRIBUTING.md (hunk 3 above)
Comment (kubukoz) [inline CONTRIBUTING.md]:
```suggestion
Serving HTTP on 0.0.0.0 port 4000 ...
```

**Comment by kubukoz** [comment]
I'll merge this on green, if there are any issues left we can always fix them later :)

**Pull request merged**